Sources/OktaAnalytics/OktaAnalytics.swift

```diff
@@ -111,21 +117,305 @@ public final class OktaAnalytics: NSObject {
+                    let scenarioProperties = try ScenarioProperty.fetchAll(db, sql: "SELECT * FROM ScenarioProperty WHERE scenarioID = \'\($0.id)\'")
+                    var properties: Dictionary = .init(uniqueKeysWithValues: scenarioProperties.lazy.map { ($0.key, $0.value) })
+                    properties["DurationMS"] = "\($0.startTime.distance(to: Date()) * 1000)"
+                    trackEvent($0.displayName.isEmpty ? $0.name : $0.displayName, withProperties: properties)
+                }
+            }
+            try databasePool?.write {
+                try $0.execute(sql: "DELETE FROM Scenario WHERE name = \'\(name)\'")
+            }
+        } catch {
+            providers.forEach {
+                $1.logger?.log(error: error as NSError, file: #file, line: #line, funcName: #function)
+            }
+        }
+    }
+
+    /**
+        end sceanrio with scenario name. clears local storage.
+        Note: events wont be reported to providers
+        - Parameters:
+           - name: scenario name
+        */
+    public static func disposeScenario(_ name: Name) {
+        lock.writeLock()
+        defer { lock.unlock() }
+        do {
+            try databasePool?.write {
+                try $0.execute(sql: "DELETE FROM Scenario WHERE name = \'\(name)\'")
+            }
+        } catch {
+            providers.forEach {
+                $1.logger?.log(error: error as NSError, file: #file, line: #line, funcName: #function)
+            }
+        }
+    }
+
+    /// Dispose Scenarios from local storage
+    /// Note: events wont be reported to providers
+    public static func disposeAllScenarios() {
+        lock.writeLock()
+        defer { lock.unlock() }
+        do {
+            try databasePool?.write {
+                try $0.execute(sql: "DELETE FROM Scenario")
+            }
+        } catch {
+            providers.forEach {
+                $1.logger?.log(error: error as NSError, file: #file, line: #line, funcName: #function)
+            }
+        }
+    }
+}
+
+extension OktaAnalytics {
+    enum DBVersions: Int, SchemaVersionType {
+        case v1 = 1
+        case v2 = 2
+    }
+
+    class SQLiteMigrator: SQLiteMigratable {
+        typealias Version = DBVersions
+        func willStartIncrementalStorageMigrationSequence(startVersion: DBVersions, endVersion: DBVersions) throws { }
+        func performIncrementalStorageMigration(_ nextVersion: DBVersions, database: Database) throws { }
+        func didFinishStorageIncrementalMigrationSequence(startVersion: DBVersions, endVersion: DBVersions) { }
+    }
+
+}
+
+// Extensions
 private extension OktaAnalytics {
+
+    static func checkAndInitializeDB() async throws -> DatabasePool? {
+        if databasePool != nil { return self.databasePool }
+        guard let cacheURL = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first else {
```

Comment:
FYI, it is local cache url. Should your solution support shared location(group containers)?